scheduler/src/cook/mesos/mesos_compute_cluster.clj

```diff
@@ -36,6 +37,39 @@
 (meters/defmeter [cook-mesos scheduler handle-status-update-rate])
 
 
+(defn conditionally-sync-sandbox
+  [conn task-id task-state sync-agent-sandboxes-fn]
+  (let [instance-ent (d/entity (d/db conn) [:instance/task-id task-id])]
+    (when (and (#{:task-starting :task-running} task-state)
+               (not= :executor/cook (:instance/executor instance-ent)))
+      ;; cook executor tasks should automatically get sandbox directory updates
+      (sync-agent-sandboxes-fn (:instance/hostname instance-ent) task-id))))
+
+(defn handle-status-update
```

Comment:
I know for historic reasons, we called this handle-status-update, but maybe we should rename it to 'write-status-update-to-datomic' for consistency with what it actually does?

Reply:
Also, is this copy&paste code from elsewhere? E.g., the predicate at line 55 inside the 'if' has the feel of being copied. Aaah, this was moved from sched/handle-status-update. But if so, why was it removed from the mesos path? Was/is it dead code for mesos?

Reply:
This is in the mesos path.

Reply:
Yeah. I was confused.